### DISCUSSION TITLE
Immutable to Mutable

### DISCUSSION THREAD

**Discussion by UYousafzai** [msgpack/msgpack-python#546] [Q&A] [answered]
Hello, sorry if this is a bit of a newbie question but I am working with msgpacks for the first time, wanted to know if the msgpack objects once unpacked are immutable objects? I wanted to modify some data while keeping the existing msgpack file and wanted to know how I would go on about achieving this?

Currently I am using another library which makes use of msgpacks and their readers are I think immutable. just wanted to know if the limitation is coming from this library or just something the subsequent library just hasn't coded.

**Comment by methane** [comment]
msgpack is like JSON. Typical unpacked data is int, float, str, bytes, list, and dict.
int, float, str, bytes are immutable. list and dict are mutable.

**Reply by UYousafzai** [reply]
I see maybe then that's a library issue, currently I am using a library and the dict and list in there are immutable, any editing done to the msgpack object reverts back to the originally read object as soon as you edit it and leave the block of code. at first I thought maybe that's because even though a python pbject msgpack read object may require a write operation to edit the object itself but I am not sure what's going wrong there if that's not the case.

To simply:
My Expected behaviour was: msgpack.read ---> msgpack (object) ---> Edit list/dict inside msgpack object ----> updated msgpack (object)
Behavior I got: msgpack.read ---> msgpack (object) ---> Edit list/dict inside msgpack object ---> original msgpack (object) edits not there.

**Reply by methane** [reply]
I don't understand what you are saying. Use small, but complete code to explain.

**Comment by UYousafzai** [comment, marked as answer]
I was able to track it, issues wasn’t with this library, it’s the subsequent library that used this.
